src/model/cloud-runner/remote-client/setup-cloud-runner-repository.ts

```diff
@@ -0,0 +1,95 @@
+import fs from 'fs';
+import CloudRunner from '../cloud-runner';
+import { CloudRunnerFolders } from '../services/cloud-runner-folders';
+import { Caching } from './caching';
+import { LFSHashing } from '../services/lfs-hashing';
+import { RemoteClientLogger } from './remote-client-logger';
+import path from 'path';
+import { assert } from 'console';
+import CloudRunnerLogger from '../services/cloud-runner-logger';
+import { CliFunction } from '../../cli/cli-functions-repository';
+import { CloudRunnerSystem } from '../services/cloud-runner-system';
+
+export class SetupCloudRunnerRepository {
+  public static async run() {
+    try {
+      await CloudRunnerSystem.Run(`mkdir -p ${CloudRunnerFolders.uniqueCloudRunnerJobFolderFull}`);
```

Comment:
`run` is quite abstract and doesn't tell much about what's happening exactly.

In this case even though the class name tries being descriptive, it's still a bit hard to find out what the run method does exactly.

In this specific case you could consider calling a `CloudRunner` method `private loadRepository()` which could in turn use a `model class` or `service` called `Repository` (part of remote-client) which could expose a descriptive method of what it's doing exactly (or a high level method that calls descriptive private methods).

Reply:
Delicious

Reply:
Renamed to `RemoteClient.bootstrapRepository`.

Within `remote-client/index.ts`